src/public/components/router.js

```diff
@@ -1,6 +1,70 @@
-import K, * as U from "karet.util"
-import React     from "karet"
+import * as R     from "ramda"
+import * as L     from "partial.lenses"
+import K, * as U  from "karet.util"
+import React      from "karet"
+import toRegex    from 'path-to-regexp'
 
-export default U.withContext(({routes, NotFound}, {path}) => U.fromKefir(K(path, path =>
-  React.createElement(routes[path] || NotFound)
-)))
+// TODO: Make it possible to add routes more than once?
+
+// begin routes initialization
+const expandRoutes = rawRoutes =>
+  R.pipe( R.toPairs
+        , R.map( R.zipObj( [ 'route', 'component' ] ) )
+        )( rawRoutes )
```

Comment:
In cases like this I would drop the argument:

```js
const expandRoutes =
   R.pipe( R.toPairs
         , R.map( R.zipObj( [ 'route', 'component' ] ) )
         )
```

This is like the ideal case for point-free programming where one can just use simple composition.